source/common/common/token_bucket_impl.cc

```diff
@@ -103,4 +104,17 @@ double AtomicTokenBucketImpl::timeNowInSeconds() const {
   return std::chrono::duration<double>(time_source_.monotonicTime().time_since_epoch()).count();
 }
 
+std::chrono::milliseconds AtomicTokenBucketImpl::nextTokenAvailable() const {
+  // If there are tokens available, return immediately.
+  if (remainingTokens() >= 1) {
+    return std::chrono::milliseconds(0);
+  }
+
+  // Calculate time since the last fill.
+  double current_time = timeNowInSeconds();
+  double last_time = time_in_seconds_.load();
+  return std::chrono::milliseconds(
+      static_cast<uint64_t>(((1 / fill_rate_ - (current_time - last_time)) * 1000)));
+}
```

Comment:
```suggestion
std::chrono::milliseconds AtomicTokenBucketImpl::nextTokenAvailable() const {
  // If there are tokens available, return immediately.
  const double remaining_tokens = remainingTokens();
  if (remaining_tokens >= 1) {
    return std::chrono::milliseconds(0);
  }

  // Calculate time since the last fill.
  return std::chrono::milliseconds(
      static_cast<uint64_t>(((1 - remaining_tokens) / fill_rate_) * 1000));
}
```